hmda/src/main/scala/hmda/api/http/filing/submissions/SubmissionHttpApi.scala

```diff
@@ -0,0 +1,149 @@
+import hmda.api.http.directives.HmdaTimeDirectives
+import akka.http.scaladsl.server.Directives._
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives._
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import hmda.messages.filing.FilingCommands.{GetFiling, GetLatestSubmission}
+import hmda.messages.institution.InstitutionCommands.GetInstitution
+import hmda.messages.submission.SubmissionCommands.CreateSubmission
+import hmda.messages.submission.SubmissionEvents.SubmissionCreated
+import hmda.model.filing.Filing
+import hmda.model.filing.submission.{Submission, SubmissionId}
+import hmda.model.institution.Institution
+import hmda.persistence.filing.FilingPersistence
+import hmda.persistence.institution.InstitutionPersistence
+import hmda.persistence.submission.SubmissionPersistence
+import hmda.api.http.codec.filing.submission.SubmissionStatusCodec._
+import io.circe.generic.auto._
+import hmda.api.http.filing.FilingResponseUtils._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+trait SubmissionHttpApi extends HmdaTimeDirectives {
+
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
+  val log: LoggingAdapter
+  implicit val ec: ExecutionContext
+  implicit val timeout: Timeout
+  val sharding: ClusterSharding
+
+  //institutions/<lei>/filings/<period>/submissions
+  val submissionCreatePath: Route =
+    path("institutions" / Segment / "filings" / Segment / "submissions") {
+      (lei, period) =>
+        timedPost { uri =>
+          val institutionPersistence =
+            sharding.entityRefFor(InstitutionPersistence.typeKey,
+                                  s"${InstitutionPersistence.name}-$lei")
+
+          val fInstitution
+            : Future[Option[Institution]] = institutionPersistence ? (
+              ref => GetInstitution(ref)
+          )
+
+          val filingPersistence =
+            sharding.entityRefFor(FilingPersistence.typeKey,
+                                  s"${FilingPersistence.name}-$lei-$period")
+
+          val filingF: Future[Option[Filing]] = filingPersistence ? (ref =>
+            GetFiling(ref))
+
+          val latestSubmissionF
+            : Future[Option[Submission]] = filingPersistence ? (ref =>
+            GetLatestSubmission(ref))
+
+          val fCheck = for {
+            i <- fInstitution
+            f <- filingF
+            l <- latestSubmissionF
+          } yield (i, f, l)
+
+          onComplete(fCheck) {
+            case Success(check) =>
+              check match {
+                case (None, _, _) =>
+                  entityNotPresentResponse("institution", lei, uri)
+                case (_, None, _) =>
+                  entityNotPresentResponse("filing", s"$lei-$period", uri)
+                case (_, _, maybeLatest) =>
+                  maybeLatest match {
+                    case None =>
+                      val submissionId = SubmissionId(lei, period, 0)
```

Comment:
Might just be a slight design question, but should the submissions be 1-based?  It might be useful when interacting with a lot of HMDA Help questions (e.g. how many submissions, what was there X submission).